app/controllers/projects/find_file_controller.rb

```diff
@@ -0,0 +1,30 @@
+# Controller for viewing a repository's file structure
+class Projects::FindFileController < Projects::ApplicationController
+  include ExtractsPath
+  include ActionView::Helpers::SanitizeHelper
+  include TreeHelper
+
+  before_action :require_non_empty_project
+  before_action :assign_ref_vars
+  before_action :authorize_download_code!
+
+  def show
+    return render_404 unless @repository.commit(@ref)
+
+    @url = namespace_project_files_path(@project.namespace, @project, @ref, @options.merge(format: :json))
+    @treeUrl = namespace_project_tree_path(@project.namespace, @project, @ref)
+    @blobUrlTemplate = namespace_project_blob_path(project.namespace, project, tree_join(@id || @commit.id))

```

Comment:
We don't need all of these in the controller, we can just use the helpers in the view.